docs/setup-guide/multi-node-dev-cluster.md

```diff
@@ -2,99 +2,147 @@
+./cluster/kubectl.sh get all --all-namespaces
+ifconfig -a
+ip route
+sudo ls -alg /etc/cni/net.d/10-flannel.conflist (for checking)
 ```
 
-If the worker node is an AWS EC2, you can download files /var/run/kubernetes/kubelet.kubeconfig and /var/run/kubernetes/client-ca.crt from the master node, and then upload to /tmp/arktos/ in the worker node.
+Note: Please reference the blog [Kubernetes: Flannel networking](https://blog.laputa.io/kubernetes-flannel-networking-6a1cb1f8ec7c) if you want to know how flannel network works in kubernetes.
 
-At the moment this doc is written, kube-proxy does not support multi-tenancy yet, and it won't be deployed in the cluster. After kube-proxy issue has been fixed (we will allocate resource to cope with it soon), probably we will also need to copy over kube-proxy related secret and artifacts.
+Note: CNI plugin of Calico is not supported because the resource 'EndpointSlices' is not supported by Arktos so far.
 
-As a temporary measure to accommodate Flannel pods to access api server through the kubernetes service IP (the notorious 10.0.0.1) at the absense of kube-proxy, we need to place a bit trick by running below command __at the worker node__, assuming the kupe-apiserver is listening at https://10.138.0.19:6443 (please substitute with the proper value of your own cluster):
+3. In the lab machine to be added as a worker node, ensure following worker secret files copied from the master node:
+If the worker node is a GCP instance  
 ```bash
-sudo iptables -t nat -A OUTPUT -p tcp -d 10.0.0.1 --dport 443 -j DNAT --to-destination 10.138.0.19:6443
+gcloud beta compute ssh --zone "us-west2-a" "doctests-worknode"  --project "workload-controller-manager"
 ```
 
-NOTE: you need to re-run the above command once you restart the machine.
+If the worker node is an AWS EC2, you can download files /var/run/kubernetes/kubelet.kubeconfig and /var/run/kubernetes/client-ca.crt from the master node, and then upload to /tmp/arktos/ in the worker node.
 
-Please be advised that this is a temporary quirk only; after we have the full service support by proper kube-proxy or other means, we don't need it any more.
+```bash
+mkdir -p /tmp/arktos
+scp -i "<private key of keypair of master node>" ubuntu@<master-node-instance>:/var/run/kubernetes/kubelet.kubeconfig /tmp/arktos/kubelet.kubeconfig
+scp -i "<private key of keypair of master node>" ubuntu@<master-node-instance>:/var/run/kubernetes/client-ca.crt /tmp/arktos/client-ca.crt
+```
 
 4. start the worker node and register into cluster
 
-First Make sure the following directories in the worker node are empty. If not, clean them up.
+First make sure the following directories in the worker node are empty. If not, clean them up using sudopermission.
 ```
-/opt/cni/bin/
-/etc/cni/net.d/
+sudo rm -rf /opt/cni/bin/*
+sudo rm -rf /etc/cni/net.d/*
 ```
 
 Then at worker node, run following commands:
 ```bash
 export ARKTOS_NO_CNI_PREINSTALLED=y
+
+hostname -i  (to get <worker-ip>)
```

Comment:
suggest to remove "to get <worker-ip>" from here. all command here suppose to be copied and run directly.

Reply:
Yes, removed "(to get <worker-ip>)". Thanks!